include/particle.tcc

```diff
@@ -242,6 +242,28 @@ void mpm::Particle<Tdim>::initialise() {
   this->properties_["displacements"] = [&]() { return displacement(); };
 }
 
+//! Assign material state variables from neighbour particle
+template <unsigned Tdim>
+bool mpm::Particle<Tdim>::assign_material_state_vars(
+    const mpm::dense_map& state_vars,
+    const std::shared_ptr<mpm::Material<Tdim>>& material) {
+  bool status = false;
+  if (material != nullptr) {
+    if (this->material_id_ == material->id() ||
+        this->material_id_ == std::numeric_limits<unsigned>::max()) {
+      // Assign material
+      // material_ = material;
```

Comment:
I wrote a test function, and I think this should be commented out. The reason is if we don't do this, we are essentially overwriting the existing material, in which case, the condition `if (mat_state_vars.size() == state_vars.size())` will always be true. I tested this with two materials with same material ids (not likely to happen, but a test case, where it will pass the first if, and it will overwrite the existing material with the new one. Since we are only interpolating new material properties, there is no need to overwrite the material pointer in the particle, what do you think? @srhgk2